mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,193 @@
+                run_id=run.info.run_id, artifact_path=artifact.path
+            )
+            if artifact.is_dir and os.path.exists(os.path.join(full_path, "MLmodel")):
+                model_paths.append(
+                    full_path
+                )  # this is the model folder, we log it after logging all artifacts
+            else:
+                self.log_artifact(context, full_path, artifact)
+        logger.debug("Finished to log MLFlow artifacts")
+        if model_paths:
+            logger.debug("Starting to log MLFlow model")
+            for path in model_paths:
+                self.log_model(path, context)
+            logger.debug("Finished to log MLFlow model")
+
+    def log_model(
+        self,
+        model_uri: str,
+        context: Union[MLClientCtx, dict],
+    ):
+
+        model_info = self._tracked_platform.models.get_model_info(model_uri=model_uri)
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            model_zip = f"{tmp_dir}-model.zip"
+        zip_folder(model_uri, model_zip)
+        key = model_info.artifact_path
+        inputs = outputs = None
+
+        if model_info.signature is not None:
+            if model_info.signature.inputs is not None:
+                inputs = schema_to_feature(
+                    model_info.signature.inputs, convert_np_dtype_to_value_type()
+                )
+            if model_info.signature.outputs is not None:
+                outputs = schema_to_feature(
+                    model_info.signature.outputs, convert_np_dtype_to_value_type()
+                )
+        context.log_model(
+            key,
+            framework="mlflow",
+            model_file=model_zip,
+            metrics=context.results,
+            parameters=model_info.flavors,
+            labels={
+                "mlflow_run_id": model_info.run_id,
+                "mlflow_version": model_info.mlflow_version,
+                "model_uuid": model_info.model_uuid,
+            },
+            extra_data=self._artifacts,
+            inputs=inputs,
+            outputs=outputs,
+        )
+
+    def log_artifact(self, context: MLClientCtx, local_path: str, artifact):
+        artifact = context.log_artifact(
+            item=pathlib.Path(artifact.path).name.replace(".", "_"),
+            local_path=local_path,
+        )
+        self._artifacts[artifact.key] = artifact
+
+    def post_run(self, context: Union[MLClientCtx, dict]):
+        import mlflow
+
+        self._client = mlflow.MlflowClient()
+        self._apply_post_run_tasks(context=context)
+
+    # todo actually implement this
+    def log_dataset(self, dataset_path, context):
+        pass
+
+
+def schema_to_feature(schema, utils) -> list:  # todo add hints here
+    """
+    changes the features from a scheme (usually tensor) to a list
+    :param schema: features as made by mlflow
+    :param utils: CommonUtils.convert_np_dtype_to_value_type, can't import here
```

Comment:
what? I'm sure this can be fixed.. if you have circular imports please fix them 🙏

Reply:
why not using the mlrun.track.util? even though I don't like that method either